File: princesspeach/5287788/peach.cc
#include <iostream>
#include <set>

int main() {
  int max, n;
  std::cin >> max >> n;

  int x;
  std::set<int> got;
  while (std::cin >> x) {
    got.insert(x);
  }

  for (int i = 0; i < max; ++i) {
    if (got.find(i) == got.end())
      std::cout << i << '\n';
  }

  std::cout << "Mario got " << got.size() << " of the dangerous obstacles.\n";

  return 0;
}

File: princesspeach/4843166/peach.py
n, y = map(int, input().split())
seen = set()

for i in range(y):
    seen.add(int(input()))

seen = sorted(seen)

prev = 0
for obstacle in seen:
    for i in range(prev, obstacle):
        print(i)
    prev = obstacle + 1

if (prev < n - 1):
    for i in range(prev, n):
        print(i)

num_seen = len(seen)
print(f'Mario got {num_seen} of the dangerous obstacles.')
